src/domain/notification/aggregate.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable max-lines-per-function */
+import type { StoreApi } from 'zustand'
+import { immer } from 'zustand/middleware/immer'
+import { createStore } from 'zustand/vanilla'
+import { devtools } from 'zustand/middleware'
+import type { IO } from 'fp-ts/lib/IO'
+import type { Reader } from 'fp-ts/lib/Reader'
+import * as A from 'fp-ts/lib/Array'
+import * as O from 'fp-ts/lib/Option'
+import { pipe } from 'fp-ts/lib/function'
+import type { ForgetType } from '@/util/type'
+import { eqNotification } from './entity'
+import type { NotificationID, NotificationType, Notifications } from './entity'
+
+type NotificationState<T> = {
+  aggregate: Notifications<T>
+}
+
+export type ReportNotificationInput<T> = {
+  id: NotificationID
+  type: NotificationType
+  title: string
+  message?: string
+  action?: T
+}
+
+export type DismissNotificationInput = {
+  id: NotificationID
+}
+
+type NotificationAction<T> = {
+  reportNotification: (input: ReportNotificationInput<T>) => IO<void>
+  dismissNotification: (input: DismissNotificationInput) => IO<void>
+  notificationWithActionReported: () => IO<boolean>
+}
+
+export type NotificationDTO<T> = {
+  id: NotificationID
+  type: NotificationType
+  title: string
+  message?: string
+  action?: T
+}
+
+export type NotificationsDTO<T> = NotificationDTO<T>[]
+
+type NotificationQuery<T> = {
+  notifications: () => IO<NotificationsDTO<T>>
+}
+
+type NotificationStore<T> = NotificationState<T> & NotificationAction<T> & NotificationQuery<T>
+export type NotificationAggregate<T> = ForgetType<NotificationState<T>, NotificationStore<T>>
```

Comment:
I would have reversed the names, because, IMHO, the `notificationAggregate` object's type should rather be a `NotificationAggregate`.

Reply:
Well, the naming is not great.
I think that to avoid any confusion, we should name it `Domain`, which is the visible part of what a domain is, and which allows to interact with an agregate (or several aggregates).

```suggestion
type NotificationStore<T> = NotificationState<T> & NotificationAction<T> & NotificationQuery<T>
export type NotificationDomain<T> = ForgetType<NotificationState<T>, NotificationStore<T>>
```